client/db/bolt/upgrades.go

```diff
@@ -247,6 +250,64 @@ func reloadMatchProofs(tx *bbolt.Tx, skipCancels bool) error {
 	})
 }
 
+// v4Upgrade moves active orders from what will become the archivedOrdersBucket
+// to a new ordersBucket. This is done in order to make searching active orders
+// faster, as they do not need to be pulled out of all orders any longer. This
+// upgrade moves active orders as opposed to inactive orders under the
+// assumption that there are less active orders to move, and so a smaller
+// database transaction occurs.
+func v4Upgrade(dbtx *bbolt.Tx) error {
+	const oldVersion = 3
+
+	if err := ensureVersion(dbtx, oldVersion); err != nil {
+		return err
+	}
+
+	// Move any inactive orders to the new archivedOrdersBucket.
+	return moveActiveOrders(dbtx)
+}
+
+// moveActiveOrders searches the v1 ordersBucket for orders that are inactive,
+// adds those to the v2 ordersBucket, and deletes them from the v1 ordersBucket,
+// which becomes the archived orders bucket.
+func moveActiveOrders(tx *bbolt.Tx) error {
+	oldOrdersBucket := []byte("orders")
+	newActiveOrdersBucket := []byte("activeOrders")
+	// NOTE: newActiveOrdersBucket created in NewDB, but TestUpgrades skips that.
+	_, err := tx.CreateBucketIfNotExists(newActiveOrdersBucket)
+	if err != nil {
+		return err
+	}
+
+	archivedOrdersBkt := tx.Bucket(oldOrdersBucket)
+	activeOrdersBkt := tx.Bucket(newActiveOrdersBucket)
+	return archivedOrdersBkt.ForEach(func(k, _ []byte) error {
+		archivedOBkt := archivedOrdersBkt.Bucket(k)
+		if archivedOBkt == nil {
+			return fmt.Errorf("order %x bucket is not a bucket", k)
+		}
+		status := order.OrderStatus(intCoder.Uint16(archivedOBkt.Get(statusKey)))
+		if status == order.OrderStatusUnknown {
+			fmt.Printf("Encountered order with unknown status: %x\n", k)
```

Comment:
Is it worth adding a `dex.Logger` argument to all the upgrade funcs?

Reply:
Let's table that. But maybe.